ompy/matrix.py

```diff
@@ -577,6 +578,125 @@ def cut_diagonal(self, E1: Optional[Iterable[float]] = None,
             matrix.values[mask] = 0.0
             return matrix
 
+    def FitPeak(self, pre_region: Tuple[float, float],
+                post_region: Tuple[float, float],
+                ex: Tuple[float,float]) -> dict[str, float]:
+        """Fit a gamma peak with a gaussian on top of a linear
+        spectra.
+        Args:
+            pre_region: Energy region with linear spectra on the left
+            side of the peak
+            post_region: Energy region with linear spectra on the right
+            side of the peak
+            ex: Either a tuple with the max/min excitation energy included
+            in the fit or the index of the excitation bin to be fitted.
+        """
+        def Gaus(x, const, mean, std):
+            return const*np.exp(-0.5*((x-mean)/std)**2)/(std*np.sqrt(2*np.pi))
+        def Pol(x, slope, intercept):
+            return x*slope + intercept
+        def Model(x, const, mean, std, slope, intercept):
+            return Gaus(x, const, mean, std) + Pol(x, slope, intercept)
+
+        pre_slice = slice(self.index_Eg(pre_region[0]), self.index_Eg(pre_region[1])+1)
+        peak_slice = slice(self.index_Eg(pre_region[1])+1, self.index_Eg(post_region[0]))
+        post_slice = slice(self.index_Eg(post_region[0]), self.index_Eg(post_region[1])+1)
+        fit_slice = slice(self.index_Eg(pre_region[0]), self.index_Eg(post_region[1])+1)
+
+        ex_slice = slice(self.index_Ex(ex[0]), self.index_Ex(ex[1])+1)
+        pre_spec = self.values[ex_slice, pre_slice].sum(axis=0)
+        peak_spec = self.values[ex_slice, peak_slice].sum(axis=0)
+        post_spec = self.values[ex_slice, post_slice].sum(axis=0)
+        fit_spec = self.values[ex_slice, fit_slice].sum(axis=0)
+
+        # Estimate the mean, std and constant
+        peak_mean = np.sum(self.Eg[peak_slice]*peak_spec)/np.sum(peak_spec)
+        peak_var = np.sum(peak_spec*self.Eg[peak_slice]**2)/np.sum(peak_spec)
+        peak_std = np.sqrt(peak_var - peak_mean**2)
+
+        # We estimate the constant from the height found at the mean
+        peak_const = np.max(peak_spec)/Gaus(peak_mean, 1., peak_mean, peak_std)
+
+        # Estimate the linear background
+        pol_estimate = np.polyfit(np.append(self.Eg[pre_slice], self.Eg[post_slice]),
+                                  np.append(pre_spec, post_spec), 1)
+
+        # Calculate the curve fitting
+        initial_guess = [peak_const, peak_mean, peak_std, pol_estimate[1], pol_estimate[0]]
+        popt, cov = curve_fit(Model, self.Eg[peak_slice], peak_spec, p0=initial_guess)
```

Comment:
use `fitslice` instead?

Reply:
You are probably correct

Reply:
If i don’t, the backgro7nd gets determined really badly. Otherwise you might fit the peak only, but fix the background